opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -777,11 +797,114 @@ mod tests {
             .load(std::sync::atomic::Ordering::Relaxed);
         assert!(is_shutdown);
 
-        processor.emit(LogData {
-            record: Default::default(),
-            instrumentation: Default::default(),
-        });
+        processor.emit(&mut log_data);
 
         assert_eq!(1, exporter.get_emitted_logs().unwrap().len())
     }
+
+    #[derive(Debug)]
+    struct FirstProcessor {
+        pub(crate) logs: Arc<Mutex<Vec<LogData>>>,
+    }
+
+    impl LogProcessor for FirstProcessor {
+        fn emit(&self, data: &mut LogData) {
+            data.record.attributes.get_or_insert(vec![]).push((
+                Key::from_static_str("processed_by"),
```

Comment:
nit: the body can also be modified, so that can be leveraged easily for testing too.

Reply:
Yes, added test for modifying body too.